realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -277,6 +277,28 @@ public String toString() {
                     ", index=" + index +
                     '}';
         }
+
+        @Override
+        public boolean equals(Object object) {
+            if (this == object) return true;
+            if (object == null || getClass() != object.getClass()) return false;
+            if (!super.equals(object)) return false;
+
+            VersionID versionID = (VersionID) object;
+
+            if (version != versionID.version) return false;
+            if (index != versionID.index) return false;
+
+            return true;
```

Comment:
I think that SharedGroup should also compare `getPath()` otherwise you risk having the same versionID across two files (highly unlikely, but still).